src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.DuplicateUserException;
+import core.basesyntax.exceptions.InvalidUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("Invalid user implementation, user == null");
+        }
+        String login = user.getLogin();
+        String password = user.getPassword();
+        int age = user.getAge();
+
+        if (login == null || password == null || age == 0) {
+            throw new InvalidUserException("");
+        }
```

Comment:
1. It's not a good practice to throw exception with empty message, even if you really wish to combine several fields to one check you should think of possible workarounds
2. Why age is compared to 0? It's also object (Integer) and can be null, so similar check is expected